### DISCUSSION TITLE
GiftCertificate in SFRA

### DISCUSSION THREAD

**Discussion by shihasmandottil** [SalesforceCommerceCloud/sfcc-ci#538] [Team Posts]
Hi,
Can anyone tell me is giftCertificate is Out of Box in SFRA. It is out of box in sitegenisis. But if it is not out of box any idea how to achieve this in SFRA.?

**Comment by pravngaur** [comment]
Hello,

No GC is not ootb sfra while it was in sg. I did it for one of the
implementations where we ended up creating a plug-in cartridge.

You will get most of the code from sg, its just the matter of hooking it up
with sfra. I would recommended creating a generic cartridge which is
independent of the gc provider(as it works in sg).

Help to help if there are any specific questions...

**Comment by syed-shahjahan** [comment]
Hi Shihas,

Currently, Gift Certificate functionality is not available in SFRA. It would be better if you create a new custom plugin so that managing of code will be easy(i.e, removing the custom code once it is available in OOTB).

At High level:
1. Render GC form
2. Create a Gift Certificate lineItem in Basket on click of "Add to Cart" button.
3. After Placing Order create a Gift Certificates by looping all GC lineItem from Order and trigger a mail.
4. Implement GC as payment.

- You can implement for Update, Delete functionality as well.

We can reuse the SiteGenesis code.

![GiftCertificatePage](https://user-images.githubusercontent.com/42941326/56141309-363c7b80-5fba-11e9-8ef3-2b8016b857bb.png)

**Comment by mdzafar** [comment]
I think currently there is no plan to support Gift certificate OOTB in SFRA, please check https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/issues/297
